azure/scope/machinepool.go

```diff
@@ -179,7 +179,7 @@ func (m *MachinePoolScope) ScaleSetSpec(ctx context.Context) azure.ResourceSpecG
 
 	spec := &scalesets.ScaleSetSpec{
 		Name:                         m.Name(),
-		ResourceGroup:                m.ResourceGroup(),
+		ResourceGroup:                m.NodeResourceGroup(),
```

Comment:
In practice it doesn't matter because AzureMachines don't work with anything else besides AzureCluster currently and the cluster scope just returns `ResourceGroup()` for `NodeResourceGroup()` but I changed everything in machine.go and machinepool.go to use `NodeResourceGroup()`.